src/client/src/app/components/register/register.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component,  OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { AppState } from 'src/app/store';
+import {
+  createUser,
+  updateUser,
+} from 'src/app/store/actions/user/user.actions';
+import { User } from '../../../../../shared/models/user.model';
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss'],
+})
+export class RegisterComponent implements OnInit {
+  addUser: FormGroup;
+  selectedUser: User | null = null;
+  constructor(
+    private fb: FormBuilder,
+    private store: Store<AppState>,
+    private router: Router,
+       
+  ) {
+    this.addUser = this.fb.group({
+      name: ['', Validators.required],
+      username: [
+        '',
+        Validators.compose([Validators.required, Validators.minLength(1)]),
+      ],
+      email: [
+        '',
+        Validators.compose([Validators.required, Validators.minLength(3)]),
+      ],
+      password: [
+        '',
+        Validators.compose([Validators.required, Validators.minLength(5)]),
+      ],
+    });
+  }
+
+  ngOnInit(): void {}
+
+  postUser(selectedUser: User | null) {
```

Comment:
personally not a fan of the name of this function or its parameter.... `registerUser(user: User | null)` is far more clear to me... BUTT i would not have a parameter on the function either way ... you can access the form inside this function with `this.adduser.value`